daemon/cmd/daemon_main.go

```diff
@@ -1436,6 +1437,21 @@ func (d *Daemon) instantiateAPI() *restapi.CiliumAPI {
 	return restAPI
 }
 
+func checkifKernelSockmapOptionsConfigured() {
+	if option.Config.SockopsEnable {
+		found := false
+		if h := probes.NewProbeManager().GetHelpers("sk_msg"); h != nil {
+			if _, ok := h["bpf_sock_hash_update"]; ok {
+				found = true
+			}
+		}
+		if !found {
+			log.Warn("BPF Sock ops not supported by kernel. Disabling the feature.")
+			option.Config.SockopsEnable = false
+		}
+	}
+}
```

Comment:
Two remarks on that function.

1. Maybe find another name? Not sure what “kernel sockmap options” refers to. Something shorter, maybe `checkSockmapSupport()` or `initSockmapOption()` or ...

2. You could get rid of the `found` variable by doing `return nil` if you find the helper. Then the `log.Warn()` does not need to be under a `if`.

Reply:
Will address it